File: LetterCombinationsofaPhoneNumber.c
#include<stdio.h>
#include<string.h>
#include<math.h>
#include<stdlib.h>
#include<stdbool.h>
char **letterCombinations(char *digits, int *returnSize)
//void letterCombinations(char *digits, int *returnSize)
{
	int lens = strlen(digits);
	returnSize = malloc(sizeof(int));
	*returnSize = 0;
	int a[8] = {3, 3, 3, 3, 3, 4, 3, 4}, l = 0, r = 1;
	char *phone[8] = {"abc", "def", "ghi", "jkl",
						"mno", "pqrs", "tuv", "wxyz"};
	char s[1000][10], **out;
	for (int i = 0; i < lens; ++ i)
	{
		int key = *(digits + i) - '2';
		for (int j = 0; j < strlen(phone[i]); ++ j)
			s[j][i] = *(phone[key] + j);
	}
	for (int i = 0; i < 10; ++ i)
		printf("%s\n", s[i]);
	
//	printf("%d\n", *returnSize);
//	*out = "abc";
//	*(out + 8) = "bcd";
//	printf("%s\n", *(out + 8));
}
int main()
{
	FILE *fin = fopen("oo.xx", "r");
	FILE *fout = fopen("xx.oo", "w");
	char *s;
	int *n;
	fscanf(fin, "%s", s);
	letterCombinations(s, n);
//	char **out = letterCombinations(s, n);
//	for (int i = 0; i < *n; ++ i)
//	{
//		printf("%s, ", out[i]);
//		fprintf(fout, "%s, ", out[i]);
//	}
	return 0;
}

File: LetterCombinationsofaPhoneNumber.py
from typing import List
import functools as ft
from functools import partial as par
def F(*z):
    z = [*z]
    z[0] = [z[0]]
    return [*ft.reduce(lambda x, y: map(y, x), z)][0]
FF = lambda *z: [*ft.reduce(lambda x, y: map(y, x), z)]
fyx = lambda f, *x: lambda *y: f(*y, *x)

class Solution:
    def generate(self, ss, s, l, r):
        if (l == r):
            if (s):
                ss.append(s)
            return ss
        for x in self.phone[int(self.digits[l]) - int('2')]:
            s += x
            ss = self.generate(ss, s, l + 1, r)
            s = s[:-1]
        return ss
    def letterCombinations(self, digits: str) -> List[str]:
        self.digits, out = digits, []
        self.phone = ["abc", "def", "ghi", "jkl",
                        "mno", "pqrs", "tuv", "wxyz"]
        out = self.generate([], '', 0, len(digits))
        return out

fin = open("oo.xx", "r")
fout = open("xx.oo", "w")
a = fin.readline()[:-1]
out = Solution().letterCombinations(a)
for x in out:
    print(x)
    fout.write(x)

